solr/core/src/java/org/apache/solr/cluster/placement/plugins/MinimizeCoresPlacementFactory.java

```diff
@@ -123,20 +123,23 @@ public List<PlacementPlan> computePlacements(
           // replicas on nodes with less cores first. We only need totalReplicasPerShard nodes given
           // that's the number of replicas to place. We assign based on the passed
           // nodeEntriesToAssign list so the right nodes get replicas.
-          ArrayList<Map.Entry<Integer, Node>> nodeEntriesToAssign =
+          List<Map.Entry<Integer, Node>> nodeEntriesToAssign =
               new ArrayList<>(totalReplicasPerShard);
-          Iterator<Map.Entry<Integer, Node>> treeIterator = nodesByCores.entries().iterator();
+          Iterator<Map.Entry<Integer, Node>> treeIterator =
+              nodesByCores.entrySet().stream()
+                  .flatMap(e -> e.getValue().stream().map(n -> Map.entry(e.getKey(), n)))
+                  .iterator();
           for (int i = 0; i < totalReplicasPerShard; i++) {
```

Comment:
Since you are adding the use of streams above, you might as well remove the need for an iterator and for loop as well.  use `Stream.limit(totalReplicasPerShard).collect(Collectors.toList())` and then the whole concoction becomes rather satisfying IMO.

Reply:
Yes much cleaner - fixed in 983a8306b78c1fa0237eb5661c2a8147a58a6a5d